File: 05_Arrays/src/Jump_Game_55.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
const canJump = function(nums) {
    let goal = nums.length-1;

    for(let i = 1; i < nums.length; i++) {
        let temp = nums.length-1-i
        if ((temp) + nums[temp] >= goal) {
            goal = temp;
        }
    }

    return goal === 0;
};

File: 05_Arrays/src/Jump_Game_55.java
public class Jump_Game_55 {
    public boolean canJump(int[] nums) {
        int goal = nums.length-1;

        for (int i = 1; i < nums.length; i++) {
            int temp = nums.length-1-i;

            if (temp + nums[temp] >= goal) {
                goal = temp;
            }
        }

        return goal == 0;
    }
}
